### PR TITLE
Feature/parent inheritance

### PR DESCRIPTION
**Pull request by Niteshvgupta**

### CODE REVIEW

File: src/Cascade.php [hunk 1]
```diff
@@ -104,25 +105,17 @@ public static function getLogger($name)
...
-            while ($last_delimit_pos !== false) {
-                $current_parent = substr($name, 0, $last_delimit_pos);
-                if (Registry::hasLogger($current_parent)) {
-                    $parent = Registry::getInstance($current_parent);
-                }
-                $last_delimit_pos = strrpos($current_parent, '.');
+        if (static::shouldInheritOnUndefined() === true) {
+            // We don't want root to inherit from root.
+            if ($name === 'root') {
+                return self::createLogger($name);
```
Comment (PhiloEpisteme) [inline src/Cascade.php:111]: This should read `static::createLogger`

File: src/Cascade.php [hunk 2]
```diff
@@ -104,25 +105,17 @@ public static function getLogger($name)
...
             }
 
-            if ($parent == null && Registry::hasLogger('default')) {
-                $parent = Registry::getInstance('default');
-            }
-            return self::createLogger($name, array(), array(), $parent);
+            $parent = self::getParent($name);
```
Comment (PhiloEpisteme) [inline src/Cascade.php:114]: Same as above, prefer `static` to `self`.

File: src/Cascade.php [hunk 3]
```diff
@@ -104,25 +105,17 @@ public static function getLogger($name)
...
-
-            // Store the pos so we only have to search through the string once
-            $last_delimit_pos = strrpos($name, '.');
-            while ($last_delimit_pos !== false) {
-                $current_parent = substr($name, 0, $last_delimit_pos);
-                if (Registry::hasLogger($current_parent)) {
-                    $parent = Registry::getInstance($current_parent);
-                }
-                $last_delimit_pos = strrpos($current_parent, '.');
+        if (static::shouldInheritOnUndefined() === true) {
```
Comment (PhiloEpisteme) [inline src/Cascade.php:108]: Where is this method defined?

File: tests/CascadeTest.php [hunk 4]
```diff
@@ -70,8 +71,8 @@ public function testGetLoggerWithInheritance()
     public function testGetLoggerWithInheritanceAndNonAlphanumericParentNames()
     {
         Cascade::setInheritOnUndefined(true);
-        $parent = Cascade::getLogger('logger_A');
-        $child = Cascade::getLogger('logger_A.child_A');
+        $parent = Cascade::getLogger('logger_Aé');
```
Comment (PhiloEpisteme) [inline tests/CascadeTest.php:74]: Looks like you're using UTF-8 chars for this one? Any reason why? Or is this a strange encoding display?

File: src/Cascade.php (hunk 3 above)
Comment (Niteshvgupta) [inline src/Cascade.php:108]: In the same class ~ line 45.

File: tests/CascadeTest.php (hunk 4 above)
Comment (Niteshvgupta) [inline tests/CascadeTest.php:74]: I wanted to make sure the logging system wouldn't break upon reading other characters allowed in PHP class names. Technically, the correct regex would be `'[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*'` but it's probably out of scope to worry about that.

**Pull request merged**